backstop/tests/carousel.html

```diff
@@ -0,0 +1,270 @@
+          <li
+            class="iui-carousel-slider-item demo-carousel-height iui-next"
+            data-slide-id="1"
+            role="group"
+            aria-roledescription="slide"
+            aria-label="2 of 3"
+            aria-hidden="true"
+          >
+            <iframe
+              width="100%"
+              height="100%"
+              src="https://www.youtube.com/embed/DT59xdKNbA8"
+              title="YouTube video player"
+              frameborder="0"
+              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+              allowfullscreen
+            ></iframe>
+          </li>
+
+          <!-- Slide with image and title. -->
+          <li
+            class="iui-carousel-slider-item demo-carousel-height iui-next"
+            data-slide-id="2"
+            role="group"
+            aria-roledescription="slide"
+            aria-label="3 of 3"
+            aria-hidden="true"
+          >
+            <div
+              class="iui-carousel-slider-item-image"
+              style="background-image: url('../assets/image-test.jpg');"
+            ></div>
+            <div class="iui-carousel-slider-item-title">Bentley System headquarters</div>
+          </li>
+
+        </ol>
+
+        <nav class="iui-carousel-navigation">
+          <button
+            class="iui-button iui-borderless iui-small iui-carousel-navigation-button iui-carousel-navigation-previous"
+            type="button"
+            aria-controls="myCarousel-items"
+            aria-label="Previous slide"
+          >
+            <svg-chevron-left
+              class="iui-button-icon"
+              aria-hidden="true"
+            ></svg-chevron-left>
+          </button>
+
+          <button
+            class="iui-button iui-borderless iui-small"
+            aria-label="Pause animation"
+            data-action="pause"
+          >
+            <svg-pause
+              class="iui-button-icon"
+              aria-hidden="true"
+            ></svg-pause>
+          </button>
+
+          <ol class="iui-carousel-navigation-dots">
+            <li>
+              <button
+                class="iui-active iui-carousel-navigation-dot"
+                data-slide="0"
+                aria-label="Slide 1 (Current slide)"
+              ></button>
+            </li>
+            <li>
+              <button
+                class="iui-carousel-navigation-dot"
+                data-slide="1"
+                aria-label="Slide 2"
+                id="test-1"
+              ></button>
+            </li>
+            <li>
+              <button
+                class="iui-carousel-navigation-dot"
+                data-slide="2"
+                aria-label="Slide 3"
+                id="test-2"
+              ></button>
+            </li>
+          </ol>
+
+          <!-- Replicates the width of the play/pause button to center the dots -->
+          <div style="width: 28px;"></div>
```

Comment:
huh, what looks hacky :/ So what is the idea if ours? We plan to allow users to pass anything in the carousel "footer"? If so, shouldnt we do something like we did with some components with 3 sections a.k.a left, center and right? I am not fan of that approach either, but I think it should be better than empty element to center things

Reply:
I have removed the empty DIV and am instead doing:
```scss
.iui-carousel-pause-button ~ .iui-carousel-navigation-button {
	margin-left: $iui-m + ($iui-xxs * 6);
}
```

Math is weird because we need it to equal 28px (same as the left / right / pause buttons).

Reply:
I don't think this is necessarily "hacky". The idea of "spacer" components is quite popular: https://www.joshwcomeau.com/react/modern-spacer-gif/

In React, I would expect a structure like this:
```tsx
<CarouselNav>
  <CarouselPrev />
  <CarouselDots />
  <CarouselNext />
</CarouselNav>
```

The user could easily add stuff in between those children.

Reply:
and who would align all of the custom content in there. We or user? In the example I am not sure I understand where the spacer would go and how we would even make reusable (for different size custom elements) ?

Reply:
Reworked the navigation area so the dots are always centered no matter how lopsided the exterior content is, no spacer `<div>` or margin required.

Reply:
Looks good, but now we have a special class for animation button.. Which has single rule for order.. That looks specific and means we want that button to be only there, nowhere else.. ANd that means, we need to have react api for that button?
idk, dont like only this part :D

Reply:
Maybe it makes sense to expose a React API for it, since the autoplay feature would require knowledge of our transforms.

Reply:
Why? I thought users just gonna switch the `activeSlide` prop and we would have animation, no??

Reply:
But I start to think us having implemented auto play function is better.. Just I really thuoght we could give user a solution to put that button (or what ever else) whenever he needs.. But maybe we can just have a flag for it and have a specific place, not sure anymore :/